server/app/models/certificate.rb

```diff
@@ -22,6 +25,27 @@ def to_path
     "#{self.grid.name}/#{self.subject}"
   end
 
+  # @return [String] with trailing newline
+  def certificate
+    pem = super
+    pem += "\n" unless pem.end_with? "\n" if pem
```

Comment:
I would still advocate just not bothering with these fixups for invalid certs, and just assuming that all certs stored in mongo are valid per the validation rules...